sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1167,6 +1210,67 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& source);
+----
+
+|Updates all of the nodes in the target graph with parameters from a
+topologically identical source graph in the modifiable state. The full
+definition of what constitutes a topologically identical graph can be found in
+the <<whole-graph-update, Whole-Graph Update>> section. Violating any of these
+topology requirements results in undefined behaviour.
+
+The characteristics in the executable graph which will be updated are detailed
+in the section on <<executable-graph-update, Executable Graph Update>>.
+
+It is not an error to update an executable graph such that all parameters of
+nodes in `graph` are identical to the arguments of the executable graph prior to
+the update.
+
+The implementation may perform a blocking wait during this call on
+any in-flight executions of that same graph if required by the backend.
+
+This function may only be called if the graph was created with the `updatable`
+property.
+
+Constraints:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::executable`.
+
+Parameters:
+
+* `source` - Modifiable graph object used as the source for updating this graph.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if the topology of
+  `source` is not the same as the target graph topology in the following ways:
+  
+** The number of nodes in the graph does not match.
+** The number of edges between nodes does not match.
+** Any node in `source` has a dependency which does not matched the paired node
+   in the target graph.
```

Comment:
Do we need to list the specifics here?  Can't we just say that an exception is thrown when the two graphs have different topologies?  We have a definition of "same topology" above, so there's no need to repeat the criteria here.

Reply:
The difference here is that these are the only ones we may reliably throw exceptions for so I included them specifically, where violating other topology requirements may be undefined behaviour. I could move this to the same section as that if that's preferable.

Reply:
What cases do you *not* diagnose with an exception?  Is it only the case when corresponding nodes have a different type?  It seems like that would be easy (and quick) to check.  No?

Reply:
Hmm, I think you are right yes, I will make this exeption definition here generic. I think what is still confusing me is whether or not we need the language around violating these rules also being undefined behaviour.

I think the undefined behaviour serves to provide some warning against the case discussed in [this comment](https://github.com/intel/llvm/pull/13253#discussion_r1635175203), but maybe that is overkill in this case. It could be replaced with language like "violating these may prevent the runtime from being able to correctly pair nodes". I suppose in my mind that leads to undefined behaviour anyway, but maybe it doesn't need to be that explicit in the spec. 

Or perhaps this wording should be removed altogether?

Reply:
Yes, I think you should choose one or the other.  Either say that `update` must throw an exception when the graphs are not topologically identical, or say that the behavior is undefined.  You shouldn't say both things.  If you say that an exception is mandated, that actually *is* defining the behavior.

Reply:
Agree that it's clearest to choose one of UB or exceptions, rather than mix them. I've updated the PR to use UB, as we can't reliably check and throw an error for all the conditions of topological identity - e.g. node ordering or edge ordering are problematic.